src/mongocxx/test/pool.cpp

```diff
@@ -183,4 +184,16 @@ TEST_CASE("a pool is created with an invalid connection string", "[pool]") {
 
     REQUIRE_THROWS_AS(pool{mongocxx::uri(uristr)}, operation_exception);
 }
+
+TEST_CASE("acquiring a client throws if waitQueueTimeoutMS expires", "[pool]") {
+    instance::current();
+    auto uri = mongocxx::uri{"mongodb://localhost:27017/?waitQueueTimeoutMS=1&maxPoolSize=1"};
+    mongocxx::pool pool{uri, options::pool(test_util::add_test_server_api())};
+    // Acquire only available client:
+    auto client = pool.acquire();
```

Comment:
```suggestion
    auto client = pool.acquire();
    CHECK(client);
```
Add assertion of the acquired pool entry (+ address unused variable warnings).